intel-sgx/ppid-retrieval-tool/main.c

```diff
@@ -0,0 +1,155 @@
+        printf("%02x", decrypted_ppid[i]); // Print each byte in hex
+    }
+    printf("\n");
+}
+
+uint32_t pce_get_pc_info(const sgx_report_t *report,
+                         const uint8_t *public_key, uint32_t key_size,
+                         uint8_t crypto_suite,
+                         uint8_t *encrypted_ppid, uint32_t encrypted_ppid_buf_size,
+                         uint32_t *encrypted_ppid_out_size,
+                         pce_info_t *pce_info,
+                         uint8_t *signature_scheme) {
+
+    sgx_launch_token_t token = {0};
+    int updated = 0;
+    sgx_status_t sgx_status = SGX_SUCCESS;
+    sgx_status_t ecall_ret = SGX_SUCCESS;
+    sgx_enclave_id_t pce_enclave_eid = 0;
+
+    if (SGX_SUCCESS != (sgx_status = sgx_create_enclave("pce/libsgx_pce.signed.so.1.25.100.1", RELEASE_ENCLAVE, &token, &updated, &pce_enclave_eid, NULL)))
+    {
+        fprintf(stderr, "Failed to create PCE enclave. The error code is:  0x%04x. \n", sgx_status);
+        sgx_status = -1;
+        goto CLEANUP;
+    }
+
+    sgx_status = get_pc_info(pce_enclave_eid,
+                             (uint32_t*) &ecall_ret,
+                             report,
+                             public_key,
+                             key_size,
+                             crypto_suite,
+                             encrypted_ppid,
+                             encrypted_ppid_buf_size,
+                             encrypted_ppid_out_size,
+                             pce_info,
+                             signature_scheme);
+
+    if (SGX_SUCCESS != sgx_status) {
+        fprintf(stderr, "Failed to call into PCE enclave: get_pc_info. The error code is: 0x%04x.\n", sgx_status);
+        sgx_status = -1;
+        goto CLEANUP;
+    }
+
+    if (SGX_SUCCESS != ecall_ret) {
+        fprintf(stderr, "Failed to get PCE info. The error code is: 0x%04x.\n", ecall_ret);
+        sgx_status = -1;
+        goto CLEANUP;
+    }
+
+    CLEANUP:
+    if(pce_enclave_eid != 0) {
+        sgx_destroy_enclave(pce_enclave_eid);
+    }
+    return sgx_status;
+}
+
+uint32_t pce_get_target_info(sgx_target_info_t *pce_target_info) {
+    sgx_launch_token_t token = {0};
+    int updated = 0;
+    sgx_status_t sgx_status = SGX_SUCCESS;
+    sgx_status_t ecall_ret = SGX_SUCCESS;
+    sgx_enclave_id_t pce_enclave_eid = 0;
+    sgx_target_info_t pce_target_info_result;
+
+    if (pce_target_info == NULL) {
+        fprintf(stderr, "Error: pce_target_info is NULL.\n");
+        sgx_status = -1;
+        goto CLEANUP;
+    }
+
+    if (SGX_SUCCESS != (sgx_status = sgx_create_enclave("pce/libsgx_pce.signed.so.1.25.100.1", RELEASE_ENCLAVE, &token, &updated, &pce_enclave_eid, NULL)))
+    {
+        fprintf(stderr, "Failed to create PCE enclave. The error code is:  0x%04x. \n", sgx_status);
+        sgx_status = -1;
+        goto CLEANUP;
+    }
+
+    if (SGX_SUCCESS != (sgx_status = sgx_get_target_info(pce_enclave_eid, &pce_target_info_result))) {
```

Comment:
You should be able to pass in the `pce_target_info` pointer directly, instead of using a local variable and then doing a `memcpy`

Reply:
This function is called from the trusted part and if you pass a pointer to the trusted part it will arrive as a NULL.